application/src/main/java/org/opentripplanner/service/vehiclerental/model/VehicleRentalStation.java

```diff
@@ -224,6 +403,338 @@ private List<RentalVehicleTypeCount> vehicleRentalTypeMapToList(
+      // we sort to have reproducible results in tests
+      .sorted(Comparator.comparing(count -> count.vehicleType().id().toString()))
       .toList();
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) return true;
+    if (o == null || getClass() != o.getClass()) return false;
+    VehicleRentalStation that = (VehicleRentalStation) o;
+    return (
+      Double.compare(that.longitude, longitude) == 0 &&
+      Double.compare(that.latitude, latitude) == 0 &&
+      isVirtualStation == that.isVirtualStation &&
+      isValetStation == that.isValetStation &&
+      vehiclesAvailable == that.vehiclesAvailable &&
+      vehiclesDisabled == that.vehiclesDisabled &&
+      spacesAvailable == that.spacesAvailable &&
+      spacesDisabled == that.spacesDisabled &&
+      isInstalled == that.isInstalled &&
+      isRenting == that.isRenting &&
+      isReturning == that.isReturning &&
+      overloadingAllowed == that.overloadingAllowed &&
+      isArrivingInRentalVehicleAtDestinationAllowed ==
+      that.isArrivingInRentalVehicleAtDestinationAllowed &&
+      realTimeData == that.realTimeData &&
+      Objects.equals(id, that.id) &&
+      Objects.equals(name, that.name) &&
+      Objects.equals(shortName, that.shortName) &&
+      Objects.equals(address, that.address) &&
+      Objects.equals(crossStreet, that.crossStreet) &&
+      Objects.equals(regionId, that.regionId) &&
+      Objects.equals(postCode, that.postCode) &&
+      Objects.equals(rentalMethods, that.rentalMethods) &&
+      Objects.equals(stationArea, that.stationArea) &&
+      Objects.equals(capacity, that.capacity) &&
+      Objects.equals(vehicleTypeAreaCapacity, that.vehicleTypeAreaCapacity) &&
+      Objects.equals(vehicleTypeDockCapacity, that.vehicleTypeDockCapacity) &&
+      Objects.equals(system, that.system) &&
+      Objects.equals(rentalUris, that.rentalUris) &&
+      Objects.equals(vehicleTypesAvailable, that.vehicleTypesAvailable) &&
+      Objects.equals(vehicleSpacesAvailable, that.vehicleSpacesAvailable) &&
+      Objects.equals(lastReported, that.lastReported)
+    );
+  }
+
+  @Override
+  public int hashCode() {
+    return Objects.hash(
+      id,
+      name,
+      shortName,
+      longitude,
+      latitude,
+      address,
+      crossStreet,
+      regionId,
+      postCode,
+      rentalMethods,
+      isVirtualStation,
+      stationArea,
+      capacity,
+      vehicleTypeAreaCapacity,
+      vehicleTypeDockCapacity,
+      isValetStation,
+      system,
+      rentalUris,
+      vehiclesAvailable,
+      vehiclesDisabled,
+      vehicleTypesAvailable,
+      spacesAvailable,
+      spacesDisabled,
+      vehicleSpacesAvailable,
+      isInstalled,
+      isRenting,
+      isReturning,
+      lastReported,
+      overloadingAllowed,
+      isArrivingInRentalVehicleAtDestinationAllowed,
+      realTimeData
+    );
+  }
+
+  public static class Builder {
```

Comment:
At a "certain size", and this is beyond the limit, we move the builders to their own files.

Reply:
Fixed in 49e18aa8cd